shared/actions/chat2/index.js

```diff
@@ -982,22 +984,27 @@ const desktopNotify = (action: Chat2Gen.DesktopNotificationPayload, state: Typed
     !meta.isMuted // ignore muted convos
   ) {
     logger.info('Sending Chat notification')
-    return Saga.put((dispatch: Dispatch) => {
-      let title = ['small', 'big'].includes(meta.teamType) ? meta.teamname : author
-      if (meta.teamType === 'big') {
-        title += `#${meta.channelname}`
-      }
-      NotifyPopup(title, {body, sound: state.config.notifySound}, -1, author, () => {
-        dispatch(
-          Chat2Gen.createSelectConversation({
-            conversationIDKey,
-            reason: 'desktopNotification',
-          })
-        )
-        dispatch(Route.switchTo([chatTab]))
-        dispatch(ConfigGen.createShowMain())
-      })
-    })
+    let title = ['small', 'big'].includes(meta.teamType) ? meta.teamname : author
+    if (meta.teamType === 'big') {
+      title += `#${meta.channelname}`
+    }
+
+    return new Promise((resolve, reject) =>
+      NotifyPopup(title, {body, sound: state.config.notifySound}, -1, author, resolve, reject)
+    )
+      .then(() =>
+        Saga.sequentially([
+          Saga.put(
+            Chat2Gen.createSelectConversation({
+              conversationIDKey,
+              reason: 'desktopNotification',
+            })
+          ),
+          Saga.put(RouteTreeGen.createSwitchTo({path: [chatTab]})),
+          Saga.put(ConfigGen.createShowMain()),
+        ])
+      )
+      .catch(_ => {})
```

Comment:
why drop errors here?

Reply:
reject is passed to onClose in notifyPopup, so it only happens if you hit close, so we do nothing

Reply:
Cool. probably worth commenting